test/core/V3/GenArt721CoreV3_Views.test.ts

```diff
@@ -1421,11 +1421,11 @@ describe("GenArt721CoreV3 Views", async function () {
   });
 
   describe("projectScriptByIndex", function () {
-    it("returns empty string by default", async function () {
-      const emptyProjectScript = await this.genArt721Core
-        .connect(this.accounts.user)
-        .projectScriptByIndex(this.projectZero, 0);
-      expect(emptyProjectScript).to.be.equal("");
+    it("reverts on invalid reads", async function () {
+      await expectRevert(
+        this.genArt721Core.projectScriptByIndex(this.projectZero, 0),
+        "ContractAsStorage: Read Error"
+      );
```

Comment:
@ryley-o want to explicitly double-click on this here and make sure you feel this behavior is acceptable, WDYT?

Reply:
hmm, we should probably actually check that they aren't checking a script index that is out of bounds in our V3 core function, and give a more informative error here. good call!

Reply:
Yeah that makes sense to me–do you think that belongs in this PR or a different follow up one? 

Alternatively, we could make behavior of `BytecodeStorage` to return `""` always in the empty-case rather than revert. 

I'm not sure I have a strong opinion here really..

Reply:
I suppose we might want to be consistent with our prior behavior, so having the contract return "" when an index is out of bounds makes sense. I thinking having the library revert is okay, since empty string should be when the bytecode of a contract represents an empty string, not when the bytecode is non-existent.

Reply:
Yeah that makes sense and SGTM–will update our implementation in the Core contract as part of this PR.

Reply:
This should be addressed now here: https://github.com/ArtBlocks/artblocks-contracts/pull/299/commits

PTAL @ryley-o and let me know if this looks off at all.

Reply:
LGTM (pretty sure you mean https://github.com/ArtBlocks/artblocks-contracts/pull/299/commits/c1ce3b85bda390647e0a003654d37fa85c1163ed) 🎉